test/ig/IGExporter.IG.test.ts

```diff
@@ -2657,4 +2657,474 @@ describe('IGExporter', () => {
+            },
+            {
+              nameUrl: 'second-nested-page.md'
+            }
+          ]
+        }
+      ];
+      r5config.parameters = [
+        {
+          code: 'copyrightyear',
+          value: '2020+'
+        },
+        {
+          code: 'generate-xml',
+          value: 'true'
+        },
+        {
+          code: 'http://example.org/parameters#special-code',
+          value: 'sparkles'
+        },
+        {
+          code: 'http://example.org/CodeSystem/sample-code-system#url-code',
+          value: 'true'
+        },
+        {
+          code: 'SampleCodeSystem#name-code',
+          value: 'true'
+        },
+        {
+          code: 'sample-code-system#id-code',
+          value: 'true'
+        }
+      ];
+
+      const pkg = new Package(r5config);
+
+      // Profile: StructureDefinition/sample-patient
+      pkg.profiles.push(
+        StructureDefinition.fromJSON(
+          fs.readJSONSync(
+            path.join(fixtures, 'profiles', 'StructureDefinition-sample-patient.json')
+          )
+        )
+      );
+
+      // Example: Patient/patient-example
+      const instanceDef = InstanceDefinition.fromJSON(
+        fs.readJSONSync(path.join(fixtures, 'examples', 'Patient-example.json'))
+      );
+      instanceDef._instanceMeta.usage = 'Example';
+      pkg.instances.push(instanceDef);
+
+      // Example: Patient/patient-example-two
+      const instanceDef2 = InstanceDefinition.fromJSON(
+        fs.readJSONSync(path.join(fixtures, 'examples', 'Patient-example-two.json'))
+      );
+      instanceDef2._instanceMeta.usage = 'Example';
+      pkg.instances.push(instanceDef2);
+
+      // Example: Patient/patient-example-three
+      const instanceDef3 = InstanceDefinition.fromJSON(
+        fs.readJSONSync(path.join(fixtures, 'examples', 'Patient-example-three.json'))
+      );
+      instanceDef3._instanceMeta.usage = 'Example';
+      pkg.instances.push(instanceDef3);
+
+      // Example: Patient/patient-example-four
+      const instanceDef4 = InstanceDefinition.fromJSON(
+        fs.readJSONSync(path.join(fixtures, 'examples', 'Patient-example-three.json'))
+      );
+      instanceDef4.id = 'patient-example-four';
+      instanceDef4._instanceMeta.name = 'patient-example-four';
+      instanceDef4._instanceMeta.usage = 'Example';
+      pkg.instances.push(instanceDef4);
+
+      // CodeSystem: CodeSystem/sample-code-system
+      const codeSystemDef = new CodeSystem();
+      codeSystemDef.id = 'sample-code-system';
+      codeSystemDef.name = 'SampleCodeSystem';
+      codeSystemDef.url = 'http://example.org/CodeSystem/sample-code-system';
+      pkg.codeSystems.push(codeSystemDef);
+
+      const exporter = new IGExporter(pkg, defs, fixtures);
+      // No need to regenerate the IG on every test -- generate it once and inspect what you
+      // need to in the tests
+      exporter.export(tempOut);
+    };
+
+    beforeAll(beforeAllR5Setup);
```

Comment:
Just curious... why did you break this into a separate function rather than just inlining it in `beforeAll`?

Reply:
I'm not 100% convinced by myself here, but the reasoning was because I had two tests for the two different `versionAlgorithm[x]` type choices. Since it was a top level property, I couldn't just put all the different cases into one list, like I did for `resources` and `pages`. So when I made those tests, they had to re-export the IG with the updated config. But after doing that, any tests that followed those tests used the latest exported IG and not the one that was exported in the `beforeAll`. So to avoid side effects and to avoid requiring those tests to be at the end of the list of tests, I called the `beforeAllR5Setup` function at the end of those two tests.

I think the same thing essentially could be accomplished by pulling the two `versionAlgorithm[x]` tests out into their own `describe` block outside of the `#r5-ig-format` block, but I decided to go with this to keep the list of tests in one group. It was a coin flip decision, so I can always go for the separate describe blocks.

Reply:
Oh.  I see.  The other option would be to put some of that (including the export) in `before` instead of `beforeAll`.  It might be a little more expensive, but then you know you don't have to ever worry about one test impacting another.

I just noticed now that you only reset the logger spy in the `beforeAll` too, so your spy is collecting all the logs from all the tests -- which could be a problem if you decide to inspect logs for any of the tests.

Reply:
Ahh yeah I did think of the `beforeEach` and was wondering if it was worth all the extra exports. I could go either way really. Do you think having the separate `beforeAllR5Setup` is too confusing? It's definitely something we don't normally do, so I would be fine re-exporting each time in a `beforeEach` if it makes it easier to understand in the future.

I might have convinced myself typing that out, but still curious what you think.

Reply:
I swapped to `beforeEach` because it probably is simpler, and at the very least, it's what we normally do. So I think ease of understanding in the future is worth the extra export calls.

And the bonus is that it resets the logger spy before each, so we can inspect logs in the test cases in the future without worrying.